### DISCUSSION TITLE
【问题】统一 Shape 的枚举

### DISCUSSION THREAD

**Discussion by teacher-zhou** [AchievedOwner/TDesignBlazor#132] [问答] [answered]
### 问题描述
`Shape` 枚举出现了很多种形式，只是某些枚举项不同，需要改为枚举类型，并根据不同组件分别定义，参考 `Theme` 枚举类型

目前发现以下组件的枚举出现重复
- ButtonShape
  * Circle
  * Square
  * Round
- ImageShape
  * Circle
  * Square
  * Round
- BadgeShape
  * Circle
  * Round
  * Dot
- TagShape
  * Round
  * Mark
  * Square
- AvatarGroup
  * Circle
  * Round

### 希望解决方案
通用的可以作为基类，调用时则是 `BadgeShape.Dot` ,`Shape.Circle`

**Comment by elvismile** [comment]
补充一个Avatar组件的Shape属性枚举，现在代码里使用的枚举是 `/Enum/Shape.cs`
AvatarShape
- Circle
- Round

**Comment by qushaoyang** [comment]
AvatarShapeEnum 这个吧

**Comment by elvismile** [comment]
我其实觉得通用性不是很强，尤其是看腾讯的这个文档变动，差异化越来越大，不如维持现状的基础上按照腾讯的文档完善一下枚举

**Comment by teacher-zhou** [comment]
> AvatarShapeEnum 这个吧

数据类型就已经能识别这个是枚举了，不需要特定的表示一个后缀，这也不符合规范
参考 [https://github.com/AchievedOwner/DotNetSpec#%E5%91%BD%E5%90%8D%E8%A7%84%E8%8C%83](https://github.com/AchievedOwner/DotNetSpec#%E5%91%BD%E5%90%8D%E8%A7%84%E8%8C%83) 的第16条

**Comment by teacher-zhou** [comment]
> 我其实觉得通用性不是很强，尤其是看腾讯的这个文档变动，差异化越来越大，不如维持现状的基础上按照腾讯的文档完善一下枚举

有道理，不需要为一个细小的内容搞成通用性，否则维护和兼容成本就变得很大

**Comment by elvismile** [comment]
> > 我其实觉得通用性不是很强，尤其是看腾讯的这个文档变动，差异化越来越大，不如维持现状的基础上按照腾讯的文档完善一下枚举
> 
> 有道理，不需要为一个细小的内容搞成通用性，否则维护和兼容成本就变得很大

那修改下issue的内容吧
1. 将枚举标准化，参照Theme的定义形式实现，每个Shape枚举单独定义实现
2. 每个枚举参照腾讯的文档（版本0.47）重新整理枚举项，保持和文档一致
3. #130 BadgeShape 按照腾讯的定义，没有Dot这个枚举项，应该是新增一个bool类型的Dot属性：文档：https://0_47_0-tdesign-vue.surge.sh/vue/components/badge?tab=api
4. #128 ButtonShape新增枚举项：Rectangle：https://0_47_0-tdesign-vue.surge.sh/vue/components/button?tab=api

**Comment by teacher-zhou** [comment]
> > > 我其实觉得通用性不是很强，尤其是看腾讯的这个文档变动，差异化越来越大，不如维持现状的基础上按照腾讯的文档完善一下枚举
> > 
> > 
> > 有道理，不需要为一个细小的内容搞成通用性，否则维护和兼容成本就变得很大
> 
> 那修改下issue的内容吧 1，将枚举标准化，参照Theme的定义形式实现，每个Shape枚举单独定义实现 2，每个枚举参照腾讯的文档（版本0.47）重新整理枚举项，保持和文档一致 3，BadgeShape 按照腾讯的定义，没有Dot这个枚举项，应该是新增一个bool类型的Dot属性：文档：https://0_47_0-tdesign-vue.surge.sh/vue/components/badge?tab=api 4，ButtonShape新增枚举项：Rectangle：https://0_47_0-tdesign-vue.surge.sh/vue/components/button?tab=api

最后两个提2个新的 issue

**Comment by teacher-zhou** [comment, marked as answer]
## 最终解决方案
1. 组件所需要的枚举由组件本身自己定义，规范命名：组件名称+Shape，例如：`ButtonShape` `AvatarShape`
2. 组件的枚举定义在组件类中，不需要特定创建一个`cs`文件
